ultralytics/nn/autobackend.py

```diff
@@ -607,6 +603,15 @@ def callback(request, userdata):
         # for x in y:
         #     print(type(x), len(x)) if isinstance(x, (list, tuple)) else print(type(x), x.shape)  # debug shapes
         if isinstance(y, (list, tuple)):
+            if (
+                (self.pb or self.engine or self.triton)
```

Comment:
Not sure if this check is required tho.

Reply:
Because it might to sufficient to just check that it's `segment` and `len(self.names) == 999`; the next two lines. It's also possible other backends have the same issue.

Reply:
Yes makes sense. Let's remove this check and put the len == 999 first.